libs/base/Data/String.idr

```diff
@@ -0,0 +1,49 @@
+module Data.String
+
+||| Convert a positive number string to a Num.
+|||
+||| ```idris example
+||| parsePositive "123"
+||| parsePositive {a=Int} " +123"
+||| ```
+parsePositive : Num a => String -> Maybe a
+parsePositive s = parsePosTrimmed (trim s) where
+  parsePosTrimmed s with (strM s)
+    parsePosTrimmed ""             | StrNil         = Nothing
+    parsePosTrimmed (strCons x xs) | (StrCons x xs) = 
+      if (x == '+') 
+        then map fromInteger (parsePosAux (unpack xs) (cast {from=Int} 0))
+        else map fromInteger (parsePosAux (unpack xs)  (cast (ord x - ord '0')))
+          where
+            parsePosAux : (List Char) -> Integer -> Maybe Integer
```

Comment:
There shouldn't be parens around `List Char` here, because application binds tighter than `->`